quiche/src/recovery/bytes_in_flight.rs

```diff
@@ -0,0 +1,144 @@
+use std::time::Duration;
+use std::time::Instant;
+
+#[derive(Default)]
+pub struct BytesInFlight {
+    // Current bytes in flight.
+    bytes_in_flight: usize,
+
+    // Instant at which bytes_in_flight transitioned from 0 to >0.
+    // Set if if bytes_in_flight is currently >0 which indicates that
+    // the bytes in flight interval is currently "open".
+    bytes_in_flight_interval_start: Option<Instant>,
+
+    // Duration of the current open interval.
+    open_interval_time: Duration,
+
+    // Sum of closed interval durations seen so far.
+    closed_interval_time: Duration,
+}
+
+impl BytesInFlight {
+    // Add to bytes in flight.  Record the start time when
+    // bytes_in_flight was 0 at the beginning of the function.
+    pub(crate) fn add(&mut self, delta: usize, now: Instant) {
+        self.bytes_in_flight += delta;
+        if delta > 0 {
```

Comment:
nit: the function only does something with an effect if `delta` in non-zero, so consider instead changing this to return early `if delta == 0` which removes a layer of code nesting